crates/polars-sql/src/context.rs

```diff
@@ -756,6 +757,69 @@ impl SQLContext {
                     polars_bail!(ComputeError: "derived tables must have aliases");
                 }
             },
+            TableFactor::UNNEST {
+                alias,
+                array_exprs,
+                with_offset,
+                with_offset_alias: _,
+            } => {
+                if let Some(alias) = alias {
+                    let table_name = alias.name.value.clone();
+                    let column_names: Vec<Option<String>> = alias
+                        .columns
+                        .iter()
+                        .map(|c| {
+                            if c.value.is_empty() {
+                                None
+                            } else {
+                                Some(c.value.clone())
+                            }
+                        })
+                        .collect();
+
+                    let column_values: Vec<Series> = array_exprs
+                        .iter()
+                        .map(|arr| parse_sql_array(arr, self))
+                        .collect::<Result<_, _>>()?;
+
+                    if column_names.is_empty() {
+                        polars_bail!(
+                            ComputeError:
+                            "UNNEST table alias must also declare column names, eg: {} (a,b,c)", alias.name.to_string()
+                        );
+                    } else if column_names.len() != column_values.len() {
+                        let plural = if column_values.len() > 1 { "s" } else { "" };
+                        polars_bail!(
+                            ComputeError:
+                            "UNNEST table alias requires {} column name{}, found {}", column_values.len(), plural, column_names.len()
+                        );
+                    }
+                    let column_series: Vec<Series> = column_values
+                        .iter()
+                        .zip(column_names.iter())
+                        .map(|(s, name)| {
+                            if let Some(name) = name {
+                                s.clone().with_name(name)
+                            } else {
+                                s.clone()
+                            }
+                        })
+                        .collect();
+
+                    let columns: Vec<_> = column_series.into_iter().map(|s| s.0.clone()).collect();
```

Comment:
This seems redundant. `DataFrame::new` can take `column_series` directly.

Reply:
🤦‍♂️